### DISCUSSION TITLE
Working with assets that keep aggregating

### DISCUSSION THREAD

**Discussion by yuval-airis** [dagster-io/dagster#18463] [Q&A]
Suppose I had a set of inputs, and tried to calculate an asset from them.
For simplicity, let's assume that the inputs are integers, and the outputs are their doubling (e.g. {1,3,72} => {2, 6, 144})

But the inputs keep being added.

Suppose I calculated the output asset once and had {2, 6, 144}, and then got a new input 12 and wanted to add to the outputs 24 too. How may I achieve that? (Should I use external storage? Can I access previous results?)

**Comment by yuval-airis** [comment]
Naively, I'd imagine something like an op that would get some triggering information and the current state and generate a new state. Is that a good way to do it?

With some pseudo-code, I'd imagine something like:
```
@asset
def counter_op(counter_op):
    if (counter_op):
       return counter_op + 1
    return 1
```

Would that work? Could two runs of it collide, and then making two increases by 1 to say 5 result in 6 instead of 7 since they'd run in parallel?

**Comment by yuhan** [comment]
You can use dynamically partition assets for this: https://docs.dagster.io/concepts/partitions-schedules-sensors/partitioning-assets#dynamically-partitioned-assets
